### PR TITLE
Replace a few #if/#elif with #ifdef and "#elif defined"

### PR DESCRIPTION
**Pull request by neikeq**
The previous code was working, but this shows intent better. Also just in case if these constants happen to be defined as `0`.

### CODE REVIEW

**Comment by aaronfranke** [comment]
Isn't this the exact opposite of your other issue, https://github.com/godotengine/godot/issues/23460?

**Comment by neikeq** [comment]
@aaronfranke Yes, but that's not how we do it in Godot currently.

File: core/vmap.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ class VMap {
 		const Pair *a = _cowdata.ptr();
 		int middle = 0;
 
-#if DEBUG_ENABLED
+#ifdef DEBUG_ENABLED
```
Comment (shartte) [inline core/vmap.h]: Shouldn't you use #if defined() here if you're using that in the other places?

Comment (neikeq) [reply core/vmap.h]: In the places I use `#if defined` is because I like it to match the `#elif`.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.1.

**Pull request merged**